### PR TITLE
Add type annotations to the project and run mypy on CI

### PR DESCRIPTION
**Pull request by jdufresne**
This helps library consumers ensure that they are using the provided API
correctly. The project includes a py.typed file for PEP-561 compliance.

This also helps ensure internal correctness and consistency. mypy also
runs on the tests, to demonstrate the test cases are representative and
real use cases.

Type checking will run on all pull requests through GitHub actions and
pre-commit.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -40,6 +40,7 @@ project_urls =
 [options]
 packages = find:
 python_requires = >=3.7
+include_package_data = true
```
[the remark this answers is not in the document]
Comment (jdufresne) [reply setup.cfg:43]: Without this line, the `py.typed` file will not be installed to the Python `site-packages` directory. 

Observe:

```
$ cd termcolor
$ pip install .
$ ls ~/.virtualenvs/termcolor/lib/python3.9/site-packages/termcolor/
... should include py.typed
```

Adjust the `ls` to be your virtualenv.

The `py.typed` file is necessary for PEP-561 compliance. More detail at:
https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-library-stubs-or-py-typed-marker

File: src/termcolor/termcolor.py [hunk 2]
```diff
@@ -89,7 +90,12 @@
 RESET = "\033[0m"
 
 
-def colored(text, color=None, on_color=None, attrs=None):
+def colored(
+    text: str,
+    color: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (jdufresne) [reply src/termcolor/termcolor.py]: Good idea! Applied this suggestion in the latest revision.

File: setup.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (hugovk) [reply setup.cfg:43]: Good stuff, thanks!

**Comment by hugovk** [comment]
Thank you!

**Pull request merged**